chromadb/utils/embedding_functions.py

```diff
@@ -479,6 +479,59 @@ def __call__(self, texts: Documents) -> Embeddings:
         return embeddings
 
 
+class VoyageAIEmbeddingFunction(EmbeddingFunction):
+    def __init__(self, api_key: str, model_name: str = "voyage-01", batch_size: int = 8):
+        """
+        Initialize the VoyageAIEmbeddingFunction.
+
+        Args:
+        api_key (str): Your API key for the HuggingFace API.
+        model_name (str, optional): The name of the model to use for text embeddings. Defaults to "voyage-01".
+        batch_size (int, optional): The number of documents to send at a time. Defaults to 8 (The max supported 3rd Nov 2023).
+        """
+        if batch_size > 8:
+            print(f"Voyage AI as of (3rd Nov 2023) has a batch size of max 8")
+
+        if not api_key:
+            raise ValueError("Please provide a VoyageAI API key.")
+
+        try:
+            import voyageai
```

Comment:
Is the source for this library available somewhere? Can you share more details about voyage ai?

Reply:
- About Voyage AI - https://docs.voyageai.com/
- Documentation of the Import / package - https://docs.voyageai.com/install

Reply:
@HammadB Its an endpoint to embed text content just like open ai text ada etc - They host their proprietary models
- You call them with your text content, you receive back the embeddings data.